ext/ManoptJuMPExt.jl

```diff
@@ -0,0 +1,321 @@
+module ManoptJuMPExt
+
+using LinearAlgebra
+import JuMP
+const MOI = JuMP.MOI
+using ManifoldsBase
+using ManifoldDiff
+
+function __init__()
+    # So that the user can use the convenient `Manopt.Optimizer`
+    setglobal!(Manopt, :Optimizer, Optimizer)
+end
+
+"""
+    struct VectorizedManifold{M} <: MOI.AbstractVectorSet
+        manifold::M
+    end
+
+Representation of points of `manifold` as a vector of `R^n` where `n` is
+`MOI.dimension(VectorizedManifold(manifold))`.
+"""
+struct VectorizedManifold{M} <: MOI.AbstractVectorSet
+    manifold::M
+end
+
+function MOI.dimension(set::VectorizedManifold)
+    return prod(ManifoldsBase.representation_size(set.manifold))
+end
+
+"""
+    Manopt.Optimizer()
+
+Creates a new optimizer object.
+
+The minimization of a function `f(X)` of an an array `X[1:n1,1:n2,...]`
+over a manifold `M` starting at `X0`, can be modeled as follows:
+```julia
+using JuMP
+@variable(model, X[i1=1:n1,i2=1:n2,...] in M, start = X0[i1,i2,...])
+@objective(model, Min, f(X))
+```
+"""
+mutable struct Optimizer <: MOI.AbstractOptimizer
+    manifold::Union{Nothing,ManifoldsBase.AbstractManifold}
+    problem::Union{Nothing,AbstractManoptProblem}
+    state::Union{Nothing,AbstractManoptSolverState}
+    variable_primal_start::Vector{Union{Nothing,Float64}}
+    sense::MOI.OptimizationSense
+    nlp_model::MOI.Nonlinear.Model
+    options::Dict{String,Any}
+    function Optimizer()
+        return new(
+            nothing,
+            nothing,
+            nothing,
+            Union{Nothing,Float64}[],
+            MOI.FEASIBILITY_SENSE,
+            MOI.Nonlinear.Model(),
+            Dict{String,Any}(),
+        )
+    end
+end
+
+MOI.get(::Optimizer, ::MOI.SolverVersion) = "0.4.37"
+
+function MOI.is_empty(model::Optimizer)
+    # TODO replace `isnothing(model.nlp_model.objective)`
+    #      by `MOI.is_empty` once the following is fixed
+    #      https://github.com/jump-dev/MathOptInterface.jl/issues/2302
+    return isnothing(model.manifold) &&
+           isempty(model.variable_primal_start) &&
+           isnothing(model.nlp_model.objective) &&
+           model.sense == MOI.FEASIBILITY_SENSE
+end
+
+function MOI.empty!(model::Optimizer)
+    model.manifold = nothing
+    model.problem = nothing
+    model.state = nothing
+    empty!(model.variable_primal_start)
+    model.sense = MOI.FEASIBILITY_SENSE
+    # TODO replace by `MOI.empty!` once the following is fixed
+    #      https://github.com/jump-dev/MathOptInterface.jl/issues/2302
+    model.nlp_model.objective = nothing
```

Comment:
```suggestion
    model.nlp_model = MOI.Nonlinear.Model()
```

Reply:
Was this change done? Then we could close this.

Reply:
Yes, it's fixed now

Reply:
The necessary feature has been added in https://github.com/jump-dev/MathOptInterface.jl/pull/2305 and this PR uses it now so @odow's suggestion is not needed anymore

Reply:
Thanks for the explanation :)